### ISSUE TITLE
Querying private emotes returns 400 with panic as an error

### ISSUE THREAD

**Issue by zneix** [SevenTV/ServerGo#8]
Emote link: https://7tv.app/emotes/60ae546c9986a00349ea35d5

Example request:

```
[zneix@uds api]$ http --verbose https://api.7tv.app/v2/gql 'query={ emote(id: "60ae546c9986a00349ea35d5") { visibility id name owner { id display_name } } }'
POST /v2/gql HTTP/1.1
Accept: application/json, */*;q=0.5
Accept-Encoding: gzip, deflate
Connection: keep-alive
Content-Length: 105
Content-Type: application/json
Host: api.7tv.app
User-Agent: HTTPie/2.4.0

{
    "query": "{ emote(id: \"60ae546c9986a00349ea35d5\") { visibility id name owner { id display_name } } }"
}


HTTP/1.1 400 Bad Request
access-control-allow-origin:
access-control-expose-headers: X-Collection-Size
content-length: 146
content-type: application/json
date: Sat, 05 Jun 2021 23:06:00 GMT
vary: Origin
x-node-name: k8s-worker-2
x-pod-internal-address: 10.244.11.179
x-pod-name: 7tv-goapi-7
x-ratelimit-limit: 100
x-ratelimit-remaining: 98
x-ratelimit-reset: 44

{
    "data": {
        "emote": null
    },
    "errors": [
        {
            "message": "panic occurred: runtime error: invalid memory address or nil pointer dereference",
            "path": [
                "emote"
            ]
        }
    ]
}
```

Also, shouldn't errors like this come back as 500 / 5xx?

**Comment by TroyDota** [comment]
unfortunately the gql implementation handles the panics, and returns an array of errors. Which limits us to know the nature of the error.

**Comment by TroyDota** [comment]
Although, this is something to look into. I have rewritten parts of the GQL implementation and i think its something i should add, the ability to detect panics at entry.

**Comment by AnatoleAM** [comment]
Fixed with 0fd9a0f3af5e547f43c59b54c8c91e2c6ab2b0b4
